ocaml/xapi/certs.ml

```diff
@@ -0,0 +1,334 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Rresult
+
+(* module [Sync] provides an interface to certificates stored on
+disk. Certificates are read upon [init], potentially manipulated, and
+written back on [sync]. No other code should write these files. Writing
+back on [sync] provides a primitive form of transactions: in case of a
+fatal error, we could avoid writing a bad state to disk. This is
+currently not actively used, though *)
+
+module Sync : sig
```

Comment:
I think this sig should be in an mli file really, because whilst the Sync module is restricted, functions added to this file in the future will not be

Reply:
I can do that. I was keeping it here while things are in flux.